sdk/src/main/scala/com/horizen/account/validation/BaseFeeBlockValidator.scala

```diff
@@ -0,0 +1,31 @@
+package com.horizen.account.validation
+
+import com.horizen.SidechainTypes
+import com.horizen.account.FeeUtils
+import com.horizen.account.block.{AccountBlock, AccountBlockHeader}
+import com.horizen.account.history.AccountHistory
+import com.horizen.account.storage.AccountHistoryStorage
+import com.horizen.validation.HistoryBlockValidator
+
+import java.math.BigInteger
+import scala.compat.java8.OptionConverters.RichOptionalGeneric
+import scala.util.Try
+
+case class BaseFeeBlockValidator() extends HistoryBlockValidator[SidechainTypes#SCAT, AccountBlockHeader, AccountBlock, AccountHistoryStorage, AccountHistory] {
+
+  override def validate(block: AccountBlock, history: AccountHistory): Try[Unit] = Try {
+    val baseFee = block.header.baseFee
+    history.getBlockById(block.header.parentId).asScala match {
+      case None =>
+        if (history.isGenesisBlock(block.id) && baseFee.compareTo(FeeUtils.INITIAL_BASE_FEE) != 0)
```

Comment:
If we do not find a block with given `parentId` the current block *must* be the genesis block, right?
Does it make sense to change this to:
- check if block is not genesis block => exception
- check if base fee differs from `INITIAL_BASEFEE` => exception